tests/test.cpp

```diff
@@ -2515,6 +2521,45 @@ void ParseUnionTest() {
           true);
 }
 
+void ValidSameNameDifferentNamespaceTest() {
+  // Duplicate table names in different namespaces must be parsable
+  flatbuffers::Parser p;
+  TEST_ASSERT(p.Parse("namespace A; table X {} namespace B; table X {}"));
+  // Duplicate union names in different namespaces must be parsable
+  flatbuffers::Parser p2;
+  TEST_ASSERT(p2.Parse("namespace A; union X {} namespace B; union X {}"));
+  // Clashing table and union names in different namespaces must be parsable
+  flatbuffers::Parser p3;
+  TEST_ASSERT(p3.Parse("namespace A; table X {} namespace B; union X {}"));
+  flatbuffers::Parser p4;
+  TEST_ASSERT(p4.Parse("namespace A; union X {} namespace B; table X {}"));
+}
+
+void MultiFileNameClashTest() {
+  auto name_clash_path =
+      flatbuffers::ConCatPathFileName(test_data_path, "name_clash_test");
+  const char *include_directories[] = { name_clash_path.c_str() };
+
+  // Load valid 2 file Flatbuffer schema
+  auto valid_path =
+      flatbuffers::ConCatPathFileName(name_clash_path, "valid_test1.fbs");
+  std::string valid_schema;
+  TEST_ASSERT(flatbuffers::LoadFile(valid_path.c_str(), false, &valid_schema));
+  // Clashing table and union names in different namespaces must be parsable
+  flatbuffers::Parser p;
```

Comment:
Better to use full variable names like `parser`, builder`, or well-known short names like `fbb` for Flatbuffers Builder.